qcodes/instrument_drivers/Keysight/keysight_b220x.py

```diff
@@ -0,0 +1,358 @@
+        """
+        KeysightB220X._available_output_ports.validate(output)
+
+        self.write(':AGND:CHAN:ENAB (@{card}01{output:02d})'.format(
+            card=self._card, output=output)
+        )
+
+    @post_execution_status_poll
+    def gnd_disable_output(self, output):
+        """
+        Removes `output` from list of ports that will be connected to GND
+        input if port is unused and bias mode is enabled.
+
+        :param output: int 1-48
+        """
+        KeysightB220X._available_output_ports.validate(output)
+
+        self.write(':AGND:CHAN:DIS (@{card}01{output:02d})'.format(
+            card=self._card, output=output)
+        )
+
+    @post_execution_status_poll
+    def gnd_enable_all_outputs(self):
+        """
+        Adds all outputs to list of ports that will be connected to GND input
+        if port is unused and bias mode is enabled.
+        """
+        self.write(':AGND:CHAN:ENAB:CARD {card}'.format(
+            card=self._card))
+
+    @post_execution_status_poll
+    def gnd_disable_all_outputs(self):
+        """
+        Removes all outputs from list of ports that will be connected to GND
+        input if port is unused and bias mode is enabled.
+        """
+        self.write(':AGND:CHAN:DIS:CARD {card}'.format(
+            card=self._card)
+        )
+
+    @post_execution_status_poll
+    def couple_port_autodetect(self):
+        """Autodetect Kelvin connections on Input ports
+
+        This will detect Kelvin connections on the input ports and enable
+        couple mode for found kelvin connections. Kelvin connections must use
+        input pairs that can be couple-enabled in order to be autodetected.
+
+        `{(1, 2), (3, 4), (5, 6), (7, 8), (9, 10), (11, 12), (13, 14)}`
+
+        Also refer to the manual for more information.
+        """
+        self.write(':COUP:PORT:DET')
+
+    def clear_status(self):
+        """Clears status register and error queue of the instrument."""
+        self.write('*CLS')
+
+    def reset(self):
+        """Performs an instrument reset.
+
+        Does not reset error queue!
+        """
+        self.write('*RST')
+
+    @post_execution_status_poll
+    def _set_connection_rule(self, mode):
+        if 'free' == self.connection_rule() and 'SROU' == mode:
+            warnings.warn('When going from *free* to *single* mode existing '
+                          'connections are not released.')
+
+        self.write(':CONN:RULE {card},{mode}'.format(card=self._card,
+                                                     mode=mode))
+
+    @post_execution_status_poll
+    def _get_connection_rule(self):
+        return self.ask(':CONN:RULE? {card}'.format(card=self._card))
+
+    @staticmethod
+    def parse_channel_list(channel_list: str):
+        """Generate a set of (input, output) tuples from a SCPI channel
+        list string.
+        """
+        pattern = r'(?P<card>\d{0,1}?)(?P<input>\d{1,2})(?P<output>\d{2})(?=(' \
```

Comment:
it's probably not a big deal here, but compiling this regexp outside of the method will increase its performance.